includes/enrolment/class-sensei-enrolment-course-calculation-job.php

```diff
@@ -153,4 +188,141 @@ private function get_query_args( $course_enrolment ) {
 
 		return $user_args;
 	}
+
+	/**
+	 * Modify user query to add the user ID check.
+	 *
+	 * @access private
+	 *
+	 * @param WP_User_Query $user_query User query to modify.
+	 */
+	public function modify_user_query_add_user_id( WP_User_Query $user_query ) {
+		global $wpdb;
+
+		$user_query->query_where .= $wpdb->prepare( ' AND ID>%d', $this->get_last_user_id() );
+	}
+
+	/**
+	 * Get the option name for tracking the current job.
+	 *
+	 * @return string
+	 */
+	private function get_current_job_option_name() {
+		return self::OPTION_TRACK_CURRENT_JOB_PREFIX . $this->course_id . '_' . $this->invalidated_only ? 1 : 0;
+	}
+
+	/**
+	 * Get the option name for tracking the last user ID.
+	 *
+	 * @return string
+	 */
+	private function get_last_user_id_option_name() {
+		$job_id = $this->get_job_id();
+
+		if ( false === $job_id ) {
+			return false;
+		}
+
+		return self::OPTION_TRACK_LAST_USER_ID_PREFIX . $job_id;
+	}
+
+	/**
+	 * Set the last calculated user ID.
+	 *
+	 * @param int $user_id User ID.
+	 */
+	public function set_last_user_id( $user_id ) {
+		$current_user_option_name = $this->get_last_user_id_option_name();
+
+		if ( ! $current_user_option_name ) {
+			return;
+		}
+
+		update_option( $current_user_option_name, (int) $user_id, false );
+	}
+
+	/**
+	 * Get the last user ID that was calculated.
+	 *
+	 * @return int
+	 */
+	public function get_last_user_id() {
+		$current_user_option_name = $this->get_last_user_id_option_name();
+
+		if ( ! $current_user_option_name ) {
+			return 0;
+		}
+
+		return (int) get_option( $current_user_option_name, 0 );
+	}
+
+	/**
+	 * Get the current job ID.
+	 *
+	 * @return string|false
+	 */
+	public function get_current_job_id() {
+		global $wpdb;
+
+		$row = $wpdb->get_row( $wpdb->prepare( "SELECT option_value FROM {$wpdb->options} WHERE option_name = %s LIMIT 1", $this->get_current_job_option_name() ) );
```

Comment:
I'm bypassing `get_option` here to avoid cache issues. I'm using the strategy in `get_option` to retrieve it by using `get_row` instead of `get_var`.